File: 661. Image Smoother/Python/Solution.py
class Solution:
    def imageSmoother(self, M):
        """
        :type M: List[List[int]]
        :rtype: List[List[int]]
        """
        m = len(M)
        n = len(M[0])
        result = [[0] * n for _ in range(m)]
        for i in range(m):
            for j in range(n):
                current = 0
                count = 0
                for a in [-1, 0, 1]:
                    for b in [-1, 0, 1]:
                        row = i + a
                        col = j + b
                        if 0 <= row < m and 0 <= col < n:
                            current += M[row][col]
                            count += 1
                result[i][j] = math.floor(current / count)
        return result

File: 661. Image Smoother/C++/Solution.cpp
class Solution {
public:
    vector<vector<int>> imageSmoother(vector<vector<int>> &M) {
        int m = M.size();
        int n = M[0].size();
        vector<vector<int>> result(m, vector<int>(n));
        int range[] = {-1, 0, 1};
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                int current = 0;
                int count = 0;
                for (int a : range) {
                    for (int b : range) {
                        int row = i + a;
                        int col = j + b;
                        if (row >= 0 && row < m && col >= 0 && col < n) {
                            current += M[row][col];
                            count++;
                        }
                    }
                }
                result[i][j] = (int) floor((double) current / count);
            }
        }
        return result;
    }
};